src/main/java/com/imsweb/datagenerator/naaccr/rule/patient/SexRule.java

```diff
@@ -20,7 +20,16 @@ public SexRule() {
     }
 
     @Override
-    public void execute(Map<String, String> record, List<Map<String, String>> otherRecords, NaaccrDataGeneratorOptions options) {
-        record.put("sex", DistributionUtils.getSex());
+    public void execute(Map<String, String> record, List<Map<String, String>> otherRecords, NaaccrDataGeneratorOptions options, Map<String, String> context) {
+
+        String wantedSex = null;
```

Comment:
First, I don't think you need to always check for a null context; you should make a pre-condition that context should never be null when calling a rule. For this implementation, you create the  context and you know it's never null; so it's more for other people using your rule. You can for example add that to the comment of the execute method in the base rule class.

All this code can then be replaced with a one-line statement:
```java
record.put("sex", context.getOrDefault("sex", DistributionUtils.getSex());
```
Like I said before, context keys should be constants. Technically the record keys should be constants as well, but there are too many of them, and we use the official field name for the NAACCR variable, so that's a bit less important.

Reply:
Fixed.